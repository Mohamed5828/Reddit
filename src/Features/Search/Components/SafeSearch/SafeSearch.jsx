import Form from "react-bootstrap/Form";
import { SafeSearchStyle, StyledSwitch } from "./SafeSearch.styled";

/**
 * Component that contains the Safe Search component.
 *
 * @Component
 * @returns {React.Component}
 */
const SafeSearch = () => {
  return (
    <SafeSearchStyle>
      <div className="safe-search-inner">
        <label htmlFor="SafeSearch">Safe Search</label>
        <StyledSwitch>
          <Form.Check type="switch" id="custom-switch" label="" />
        </StyledSwitch>
      </div>
    </SafeSearchStyle>
  );
};

export default SafeSearch;
