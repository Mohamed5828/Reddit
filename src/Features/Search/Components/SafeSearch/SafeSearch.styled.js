import styled from "styled-components";
import Form from "react-bootstrap/Form";
export const SafeSearchStyle = styled.div`
  margin-left: auto;
  @media (max-width: 400px) {
    margin: auto;
    margin-bottom: 24px;
  }
  & .safe-search-inner {
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    display: flex;
    align-items: center;
    color: ${({ theme }) => theme.color.titleText_of_link};
  }
  & .safe-search-inner label {
    margin-right: 30px;
  }
`;

export const StyledSwitch = styled(Form)`
  & .form-check-input {
    font-size: 20px;
  }
  & .form-switch {
    display: flex;
    align-items: center;
  }
  & label {
    display: none;
  }
  & .form-check {
    margin-bottom: 5px;
  }
`;
