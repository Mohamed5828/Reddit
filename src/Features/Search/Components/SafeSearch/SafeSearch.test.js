import { shallow } from "enzyme";
import SafeSearch from "./SafeSearch";
describe("Safe Search", () => {
  it("this is a test for Safe Search", () => {
    expect(shallow(<SafeSearch />)).toMatchSnapshot();
  });
});
