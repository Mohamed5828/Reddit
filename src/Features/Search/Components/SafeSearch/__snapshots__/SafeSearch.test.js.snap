// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Safe Search this is a test for Safe Search 1`] = `
<styled.div>
  <div
    className="safe-search-inner"
  >
    <label
      htmlFor="SafeSearch"
    >
      Safe Search
    </label>
    <Styled(Form)>
      <FormCheck
        id="custom-switch"
        label=""
        type="switch"
      />
    </Styled(Form)>
  </div>
</styled.div>
`;
